x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/trained_models_deployment_modal.tsx

```diff
@@ -8,179 +8,175 @@ import { EuiConfirmModal, useGeneratedHtmlId, EuiHealth } from '@elastic/eui';
-            'xpack.idxMgmt.indexDetails.trainedModelsDeploymentModal.textTrainedModelManagementLink',
-            {
-              defaultMessage: 'Go to Trained Model Management',
-            }
-          )}
-        </EuiLink>
-      </EuiConfirmModal>
-    );
+  const startModelAllocation = async (trainedModelId: string) => {
+    try {
+      await ml?.mlApi?.trainedModels.startModelAllocation(trainedModelId);
+    } catch (error) {
+      setErrorsInTrainedModelDeployment((previousState) => ({
+        ...previousState,
+        [trainedModelId]: error.message,
+      }));
+      showErrorToasts(error);
+      setIsModalVisible(true);
+    }
   };
 
-  const ErroredModelsDeploymentModal = () => {
-    const pendingDeploymentsList = pendingDeployments.map((deployment, index) => (
-      <li key={index}>
-        <EuiHealth textSize="xs" color="danger">
-          {deployment}
-        </EuiHealth>
-      </li>
-    ));
+  useEffect(() => {
+    const models = inferenceIdsInPendingList.map(
+      (inferenceId) => inferenceToModelIdMap?.[inferenceId]
+    );
+    for (const model of models) {
+      if (model && !model.isDownloading && !model.isDeployed) {
+        // Sometimes the model gets stuck in a ready to deploy state, so we need to trigger deployment manually
+        startModelAllocation(model.trainedModelId);
+      }
+    }
+    const pendingModels = models
+      .map((model) => {
+        return model?.trainedModelId && !model?.isDeployed ? model?.trainedModelId : '';
+      })
+      .filter((trainedModelId) => !!trainedModelId);
+    const uniqueDeployments = pendingModels.filter(
+      (deployment, index) => pendingModels.indexOf(deployment) === index
+    );
+    setPendingDeployments(uniqueDeployments);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [inferenceIdsInPendingList, inferenceToModelIdMap]);
+
+  const erroredDeployments = pendingDeployments.filter(
+    (deployment) => errorsInTrainedModelDeployment[deployment]
+  );
 
-    return (
-      <EuiConfirmModal
-        aria-labelledby={modalTitleId}
-        style={{ width: 600 }}
-        title={i18n.translate(
-          'xpack.idxMgmt.indexDetails.trainedModelsDeploymentModal.deploymentErrorTitle',
-          {
-            defaultMessage: 'Models could not be deployed',
-          }
-        )}
-        titleProps={{ id: modalTitleId }}
-        onCancel={closeModal}
-        onConfirm={refreshModal}
-        cancelButtonText={i18n.translate(
-          'xpack.idxMgmt.indexDetails.trainedModelsDeploymentModal.deploymentErrorCancelButtonLabel',
-          {
-            defaultMessage: 'Cancel',
-          }
+  useEffect(() => {
+    if (erroredDeployments.length > 0 || pendingDeployments.length > 0) {
+      setIsModalVisible(true);
+    }
+  }, [erroredDeployments.length, pendingDeployments.length]);
+  return isModalVisible ? (
+    <EuiConfirmModal
+      aria-labelledby={modalTitleId}
+      style={{ width: 600 }}
```

Comment:
nit: I think we could use EuiThemeBreakpoints in order to avoid hardcoding widths like this

Reply:
I agree, but this was existing code I didn't want to touch too much :)